### PR TITLE
Scrolling / parallax options on the expandable commercial unit

### PR DESCRIPTION
**Pull request by austinh7**
@sndrs for fastdom help

This adds three options to the "expandable" ad unit.

When used it adds three scrolling options for the background on the expandable.

We have:
1. Parallax option (Simple Parallax scroll).
   expandablescrollpa
2. Split (Background "drops in" for top third, stays still in middle third and "drops out" for bottom third.
   expandablescrollspl
3. Fixed (Same as current scrolling MPU).
   expandablescrollfix

### CODE REVIEW

File: static/src/javascripts/projects/common/modules/commercial/creatives/expandable-v2.js [hunk 1]
```diff
@@ -39,18 +41,70 @@ define([
...
-            this.$button.toggleClass('button-spin');
-            $('.ad-exp__open-chevron').toggleClass('chevron-down');
-            this.$ad.css('height', this.openedHeight);
-            this.isClosed = false;
-            return true;
+    /**
+     * TODO: rather blunt instrument this, due to the fact *most* mobile devices don't have a fixed
+     * background-attachment - need to make this more granular
+     */
+    ExpandableV2.hasScrollEnabled = !detect.isIOS() && !detect.isAndroid();
```
Comment (Calanthe) [inline static/src/javascripts/projects/common/modules/commercial/creatives/expandable-v2.js]: what about browsers on mobile? like detect.getBreakpoint() !== 'mobile'

**Comment by uplne** [comment]
It's very difficult to read but nice use of fastdom :+1:

File: static/src/javascripts/projects/common/modules/commercial/creatives/expandable-v2.js [hunk 2]
```diff
@@ -39,18 +41,70 @@ define([
...
+
+    ExpandableV2.prototype.updateBgPosition = function () {
+        var adHeight, inViewB, inViewT, topCusp, bottomCusp, bottomScroll, topScroll;
+        fastdom.read(function () {
+            adHeight = (this.isClosed) ?
```
Comment (Calanthe) [inline static/src/javascripts/projects/common/modules/commercial/creatives/expandable-v2.js]: are those brackets needed?

File: static/src/javascripts/projects/common/modules/commercial/creatives/expandable-v2.js (hunk 1 above)
Comment (austinh7) [inline static/src/javascripts/projects/common/modules/commercial/creatives/expandable-v2.js]: This is the same functionality as the scrollable MPU, we could look at updating it all at the same time at some point?

File: static/src/javascripts/projects/common/modules/commercial/creatives/expandable-v2.js [hunk 3]
```diff
@@ -77,29 +131,47 @@ define([
                 showPlus: (this.params.showMoreType === 'plus-only' || this.params.showMoreType === 'plus-and-arrow') ?
                     '<button class="ad-exp__close-button ad-exp__open">' + svgs('closeCentralIcon') + '</button>' : ''
             },
-            $expandablev2 = $.create(template(expandableV2Tpl, _.merge(this.params, showmoreArrow, showmorePlus, videoDesktop)));
+            scrollingbg = {
+                scrollbg: (this.params.backgroundImagePType !== '' || this.params.backgroundImagePType !== 'none') ?
+                    '<div class="ad-exp--expand-scrolling-bg" style="background-image: url(' + this.params.backgroundImageP + '); background-position: ' + this.params.backgroundImagePPosition + ' 50%;"></div>' : ''
```
Comment (Calanthe) [inline static/src/javascripts/projects/common/modules/commercial/creatives/expandable-v2.js]: I think that this:
$.create('<div>')
  .addClass('foo')
  .css({
    color: 'red',
    'background-color': 'white'
  })

is more readable that your one liner. But I am not sure if it is worth to update it everywhere.

**Pull request merged**